### PR TITLE
Revamp vsbond auction

### PR DESCRIPTION
**Pull request by herryho**

### CODE REVIEW

File: pallets/vsbond-auction/src/lib.rs [hunk 1]
```diff
@@ -443,11 +547,12 @@ pub mod pallet {
 				);
 
 				if new_order_info.order_type == OrderType::Buy {
-					T::MultiCurrency::unreserve(
-						token_owner,
+					T::MultiCurrency::transfer(
```
Comment (AllenPocketGamer) [inline pallets/vsbond-auction/src/lib.rs:573]: There will occur an error when touching the `ExistDeposit` Limit;

File: pallets/vsbond-auction/src/lib.rs [hunk 2]
```diff
@@ -500,6 +631,41 @@ pub mod pallet {
...
+			let order_info = Self::order_info(order_id).ok_or(Error::<T, I>::NotFindOrderInfo)?;
+
+			let (token_to_return, amount_to_return) = match order_info.order_type {
+				OrderType::Buy => (T::InvoicingCurrency::get(), order_info.remain_price),
+				OrderType::Sell => (order_info.vsbond, order_info.remain),
+			};
+
+			// To transfer back the unused amount
+			let module_account: AccountIdOf<T> = T::PalletId::get().into_account();
+			T::MultiCurrency::transfer(
```
Comment (AllenPocketGamer) [inline pallets/vsbond-auction/src/lib.rs:682]: Again~, there will occur an error when touching `ExistDeposit` limit;

**Comment by herryho** [comment]
@AllenPocketGamer The existential deposit transfer problem has been fixed. Please review.

**Review by AllenPocketGamer — COMMENTED** [review]
ok, there doesn't seem to be a problem

**Review by ark930 — APPROVED** [review]

**Pull request merged**